src/tests/java/htsjdk/samtools/util/IntervalListTest.java

```diff
@@ -469,4 +470,33 @@ public void testMerges() {
         Assert.assertEquals(out.getStart(), 500);
         Assert.assertEquals(out.getEnd(), 699);
     }
+
+    @Test
+    public void testBreakAtBands() {
```

Comment:
please also test for 
[start=11 end=10] with band == 10
[start=10 end=9] with band == 10